.github/workflows/run-tests.yaml

```diff
@@ -3,7 +3,8 @@ name: Integration Tests
 on:
   push:
     branches: [ "master" ]
-  pull_request:
+  pull_request:  
+  workflow_call:
```

Comment:
so, I have a suggestion here that might not be strict related to the PR: instead of enabling the workflow to run every push on master, you could create a rule enabling commits on master **only** by pull requests. Safer approach in my opinion.

Reply:
> so, I have a suggestion here that might not be strict related to the PR: instead of enabling the workflow to run every push on master, you could create a rule enabling commits on master **only** by pull requests. Safer approach in my opinion.

Yep, I also think that is a safer approach. It would also justify @heitorPB 's comment.